File: srcs/texte.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   texte.c                                            :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: becaraya <becaraya@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2019/12/12 20:51:16 by becaraya          #+#    #+#             */
/*   Updated: 2019/12/13 21:34:29 by becaraya         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include "doom_nukem.h"

void	init_text_edit(t_al *al)
{
	set_text(&al->text.sect_index, "WHO", get_rect(25, 20),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.wall_index, "WHO WALL", get_rect(25, 58),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.x_start, "X1", get_rect(25, 100),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.y_start, "Y1", get_rect(150, 100),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.x_end, "X2", get_rect(25, 150),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.y_end, "Y2", get_rect(150, 150),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.cancel, "CANCEL", get_rect(600, 20),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.settings, "Settings", get_rect(25, 200),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.sector, "Sector", get_rect(70, 250),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.fl_tex, "Floor Texture", get_rect(70, 330),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.fl_hei, "Floor Height", get_rect(70, 410),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.ce_tex, "Ceiling Texture", get_rect(300, 330),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	init_text_edit_next(al);
}

void	init_text_edit_next(t_al *al)
{
	set_text(&al->text.ce_hei, "Ceiling Height", get_rect(300, 410),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.wall, "Wall", get_rect(300, 250),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.wall_tex, "Wall Texture", get_rect(70, 330),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.tools, "Tools", get_rect(25, 500),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.set_spawn, "Set Spawn", get_rect(70, 550),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.set_bad_pig, "Set Goret", get_rect(300, 550),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.draw, "Draw", get_rect(520, 550),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.link, "Sector Link", get_rect(70, 615),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
	set_text(&al->text.gen_map, "GENERATION MAP", get_rect(295, 665),
	add_color(TEXT_EDITOR)) == -1 ? yeet(al) : 0;
}

File: web version/src/texte.js
function initTextEdit(al) {
    setText(al.text.sectIndex, "WHO", getRect(25, 20), addColor(TEXT_EDITOR));
    setText(al.text.wallIndex, "WHO WALL", getRect(25, 58), addColor(TEXT_EDITOR));
    setText(al.text.difEz, "Difficulty : Easy", getRect(70, 135), addColor(TEXT_EDITOR));
    setText(al.text.difHa, "Difficulty : Hard", getRect(70, 135), addColor(TEXT_EDITOR));
    setText(al.text.cancel, "CANCEL", getRect(600, 20), addColor(TEXT_EDITOR));
    setText(al.text.settings, "Settings", getRect(25, 200), addColor(TEXT_EDITOR));
    setText(al.text.sector, "Sector", getRect(70, 250), addColor(TEXT_EDITOR));
    setText(al.text.flTex, "Floor Texture", getRect(70, 330), addColor(TEXT_EDITOR));
    setText(al.text.flHei, "Floor Height", getRect(70, 410), addColor(TEXT_EDITOR));
    setText(al.text.ceTex, "Ceiling Texture", getRect(300, 330), addColor(TEXT_EDITOR));
    initTextEditNext(al);
}

function initTextEditNext(al) {
    setText(al.text.ceHei, "Ceiling Height", getRect(300, 410), addColor(TEXT_EDITOR));
    setText(al.text.ceHeiNum, "Ceiling Hei Num", getRect(570, 410), addColor(TEXT_EDITOR));
    setText(al.text.flHeiNum, "Floor Hei Num", getRect(570, 410), addColor(TEXT_EDITOR));
    setText(al.text.wall, "Wall", getRect(300, 250), addColor(TEXT_EDITOR));
    setText(al.text.wallTex, "Wall Texture", getRect(70, 330), addColor(TEXT_EDITOR));
    setText(al.text.tools, "Tools", getRect(25, 500), addColor(TEXT_EDITOR));
    setText(al.text.setSpawn, "Set Spawn", getRect(70, 550), addColor(TEXT_EDITOR));
    setText(al.text.setEnd, "Set End", getRect(70, 550), addColor(TEXT_EDITOR));
    setText(al.text.setBadPig, "Set Goret", getRect(300, 550), addColor(TEXT_EDITOR));
    setText(al.text.draw, "Draw", getRect(520, 550), addColor(TEXT_EDITOR));
    setText(al.text.link, "Sector Link", getRect(70, 615), addColor(TEXT_EDITOR));
    setText(al.text.genMap, "GENERATION MAP", getRect(295, 665), addColor(TEXT_EDITOR));
    initTextEditNextNext(al);
}

function initTextEditNextNext(al) {
    setText(al.text.setPlayer, "Player", getRect(520, 250), addColor(TEXT_EDITOR));
    setText(al.text.life, "Life", getRect(70, 330), addColor(TEXT_EDITOR));
    setText(al.text.power, "Power", getRect(300, 330), addColor(TEXT_EDITOR));
    setText(al.text.weapon, "Weapon", getRect(520, 330), addColor(TEXT_EDITOR));
    setText(al.text.size, "Size", getRect(70, 410), addColor(TEXT_EDITOR));
    setText(al.text.mass, "Mass", getRect(300, 410), addColor(TEXT_EDITOR));
    setText(al.text.playerValue, "Player num", getRect(570, 410), addColor(TEXT_EDITOR));
    setText(al.text.resetMap, "Reset Map", getRect(300, 135), addColor(TEXT_EDITOR));
    setText(al.text.resetPlayer, "Reset Player", getRect(515, 135), addColor(TEXT_EDITOR));
    setText(al.text.gravity, "Gravity", getRect(300, 615), addColor(TEXT_EDITOR));
    setText(al.text.gNum, "9.81", getRect(570, 625), addColor(TEXT_EDITOR));
    setText(al.text.isDoor, "Is Door", getRect(520, 330), addColor(TEXT_EDITOR));
}

function setText(textObj, text, rect, color) {
    textObj.str = text;
    textObj.rect = rect;
    textObj.color = color;
}

function getRect(x, y) {
    return { x, y, w: 0, h: 0 };
}

function addColor(color) {
    return { r: (color >> 16) & 0xFF, g: (color >> 8) & 0xFF, b: color & 0xFF, a: (color >> 24) & 0xFF };
}